### PR TITLE
Use `bun` for docs instead of pnpm

### PR DESCRIPTION
**Pull request by pat-s**
Superseedes #3676 #3440 

Discussion ref #2769 

Building and serving locally worked.

### CODE REVIEW

**Comment by pat-s** [comment]
@anbraten Not sure if you have more experience here, but `bun` is working just fine locally (`bun i && bun run build`) and is 10x faster then `pnpm`.

I don't see why CI complains about `docusaurus/faster` being missing, it is listed in the config file and properly added in the lockfiles.

**Comment by anbraten** [comment]
Bun is locally sometimes using just node underneath could you try `bun --bun`

File: docs/package.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
   "private": true,
   "scripts": {
     "start": "cd ../ && make docs && cd docs && docusaurus start",
-    "build": "pnpm build:woodpecker-plugins && docusaurus build",
-    "build:woodpecker-plugins": "cd plugins/woodpecker-plugins && pnpm i && pnpm build",
+    "build": "bun build:woodpecker-plugins && docusaurus build",
+    "build:woodpecker-plugins": "cd plugins/woodpecker-plugins && bun install && bun pm trust --all && bun run build",
```
Comment (anbraten) [inline docs/package.json:8]: Should probably stick to workspaces instead

**Comment by pat-s** [comment]
Can't find a solution for the CI errors, locally everything works 🤷️

**Pull request closed without merging**